File: java/parser/src/main/java/gov/lbl/parser/parser/SubscriptVisitor.java
package gov.lbl.parser.parser;

import gov.lbl.antlr4.visitor.modelicaBaseVisitor;
import gov.lbl.antlr4.visitor.modelicaParser;
import gov.lbl.parser.domain.Expression;
import gov.lbl.parser.domain.Subscript;

public class SubscriptVisitor extends modelicaBaseVisitor<Subscript> {
    @Override
    public Subscript visitSubscript(modelicaParser.SubscriptContext ctx) {
        Boolean colon = ctx.SYMBOL_COLON() == null ? false : true;
        ExpressionVisitor expressionVisitor = new ExpressionVisitor();
    	Expression expression = ctx.expression() == null ? null : ctx.expression().accept(expressionVisitor);

        return new Subscript(expression, colon);
    }
}

File: jsParser/parser/SubscriptVisitor.js
const modelicaVisitor = require('../antlrFiles/modelicaVisitor').modelicaVisitor
const Subscript = require('../domain/Subscript')

const ExpressionVisitor = require('./ExpressionVisitor')

class SubscriptVisitor {
  constructor () {
    modelicaVisitor.call(this)
    return this
  }

  visitSubscript (ctx) {
    let expression = null
    const color_op = !!ctx.SYMBOL_COLON()

    if (ctx.expression()) {
      const expressionVisitor = new ExpressionVisitor.ExpressionVisitor()
      expression = expressionVisitor.visitExpression(ctx.expression())
    }
    return new Subscript.Subscript(expression, color_op)
  }
};

SubscriptVisitor.prototype = Object.create(modelicaVisitor.prototype)

exports.visitSubscript = this.visitSubscript
exports.SubscriptVisitor = SubscriptVisitor
